PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/ConnectionRunnerForRunConfiguration.java

```diff
@@ -89,20 +90,26 @@ public boolean execute(@NotNull DataContext dataContext, @NotNull RunConfigurati
         public boolean isApplicableFor(@NotNull RunConfigurationBase<?> configuration) {
             final boolean applicable = configuration.getProject().getService(ConnectionManager.class)
                     .getConnections().stream().anyMatch(c -> c.isApplicableFor(configuration));
-            if (applicable) {
-                if (configuration.getBeforeRunTasks().stream().noneMatch(t -> t instanceof MyBeforeRunTask)) {
-                    final MyBeforeRunTask task = new MyBeforeRunTask();
-                    task.setEnabled(true);
-                    configuration.getBeforeRunTasks().add(task);
-                }
-            } else {
-                final List<BeforeRunTask<?>> tasks = configuration.getBeforeRunTasks().stream()
-                        .filter(t -> t instanceof MyBeforeRunTask).collect(Collectors.toList());
-                if (tasks.size() > 0) {
-                    configuration.getBeforeRunTasks().removeAll(tasks);
-                }
+            final List<BeforeRunTask<?>> tasks = configuration.getBeforeRunTasks();
+            final List<BeforeRunTask<?>> myTasks = tasks.stream().filter(t -> t instanceof MyBeforeRunTask).collect(Collectors.toList());
+            if (applicable && myTasks.isEmpty()) {
+                final MyBeforeRunTask task = new MyBeforeRunTask();
+                task.setEnabled(true);
+                this.addTask(configuration, task);
+            } else if (!myTasks.isEmpty()) {
+                tasks.removeAll(myTasks);
             }
```

Comment:
a little comments here? 
From above code lines, I don't know - why is it "addTask" when myTasks.isEmpty and "removeAll" when myTasks.isNotEmpty ?

Reply:
it means there is a applicable resource connection for this run configuration when  `applicable==true`, which means we need to add a `MyBeforeRunTask` to this run configuration.
so when `applicable && myTasks.isEmpty`, we need to add a before run task.
but when **NOT** `applicable`, we need to remove the task.